Dockerfile

```diff
@@ -1,7 +1,145 @@
+jdk.unsupported,\
+# add additional cipher suites
+jdk.crypto.cryptoki,\
+# add ability to open ZIP/JAR files
+jdk.zipfs,\
+# Elasticsearch 7+ crashes without Thai Segmentation support
+jdk.localedata --include-locales en,th \
+    --compress 2 --strip-debug --no-header-files --no-man-pages --output /usr/lib/jvm/java-11 && \
+  cp ${JAVA_HOME}/bin/javac /usr/lib/jvm/java-11/bin/javac && \
+  cp -r ${JAVA_HOME}/include /usr/lib/jvm/java-11/include && \
+  mv /usr/lib/jvm/java-11/lib/modules /usr/lib/jvm/java-11/lib/modules.bk; \
+  cp -r ${JAVA_HOME}/lib/* /usr/lib/jvm/java-11/lib/; \
+  mv /usr/lib/jvm/java-11/lib/modules.bk /usr/lib/jvm/java-11/lib/modules; \
+  rm -rf /usr/bin/java ${JAVA_HOME} && ln -s /usr/lib/jvm/java-11/bin/java /usr/bin/java
+ENV JAVA_HOME /usr/lib/jvm/java-11
+ENV ES_JAVA_HOME /usr/lib/jvm/java-11
+ENV PATH "${PATH}:${JAVA_HOME}/bin"
+
+# Install Maven - taken from official repo:
+# https://github.com/carlossg/docker-maven/blob/master/openjdk-11/Dockerfile)
+ARG MAVEN_VERSION=3.6.3
+ARG USER_HOME_DIR="/root"
+ARG SHA=26ad91d751b3a9a53087aefa743f4e16a17741d3915b219cf74112bf87a438c5
+ARG BASE_URL=https://apache.osuosl.org/maven/maven-3/${MAVEN_VERSION}/binaries
+RUN mkdir -p /usr/share/maven /usr/share/maven/ref \
+  && curl -fsSL -o /tmp/apache-maven.tar.gz ${BASE_URL}/apache-maven-$MAVEN_VERSION-bin.tar.gz \
+  && echo "${SHA}  /tmp/apache-maven.tar.gz" | sha256sum -c - \
+  && tar -xzf /tmp/apache-maven.tar.gz -C /usr/share/maven --strip-components=1 \
+  && rm -f /tmp/apache-maven.tar.gz \
+  && ln -s /usr/share/maven/bin/mvn /usr/bin/mvn
+ENV MAVEN_HOME /usr/share/maven
+ENV MAVEN_CONFIG "$USER_HOME_DIR/.m2"
+ADD https://raw.githubusercontent.com/carlossg/docker-maven/master/openjdk-11/settings-docker.xml /usr/share/maven/ref/
+
+# set workdir
+RUN mkdir -p /opt/code/localstack
+WORKDIR /opt/code/localstack/
+
+# https://github.com/pires/docker-elasticsearch/issues/56
+ENV ES_TMPDIR /tmp
+
+# install npm dependencies
+ADD localstack/package.json localstack/package.json
+RUN cd localstack && npm install && rm -rf /root/.npm;
+
+# install supervisor
+RUN pip3 install supervisor
+
+# init environment and cache some dependencies
+ARG DYNAMODB_ZIP_URL=https://s3-us-west-2.amazonaws.com/dynamodb-local/dynamodb_local_latest.zip
+RUN mkdir -p /opt/code/localstack/localstack/infra && \
+    mkdir -p /opt/code/localstack/localstack/infra/dynamodb && \
+      curl -L -o /tmp/localstack.ddb.zip ${DYNAMODB_ZIP_URL} && \
+      (cd localstack/infra/dynamodb && unzip -q /tmp/localstack.ddb.zip && rm /tmp/localstack.ddb.zip) && \
+    curl -L -o /tmp/elasticmq-server.jar \
+        https://s3-eu-west-1.amazonaws.com/softwaremill-public/elasticmq-server-1.1.0.jar
+
+# installing terraform temporary - removed in the final docker image
+# TODO: Remove $PATH variable from the final docker image
+ARG TERRAFORM_ZIP_URL=https://releases.hashicorp.com/terraform/0.13.5/terraform_0.13.5_linux_${TARGETARCH}.zip
```

Comment:
FYI @alexrashed @chezsmithy - we had to upgrade to a newer version of Terraform, to fix recent build issues in our CI pipeline (see change [here](https://github.com/localstack/localstack/commit/d03da7f0bc40b6c6525d35cb50ce1ecb4667f9cd)).

We should be able to replace this with:
```
ARG TERRAFORM_ZIP_URL=https://releases.hashicorp.com/terraform/0.15.5/terraform_0.15.5_linux_amd64.zip
```